src/mscorlib/src/System/Environment.cs

```diff
@@ -701,20 +690,19 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
             }
             else
             {
-                throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)target));
+                throw new ArgumentOutOfRangeException(nameof(target), target, SR.Format(SR.Arg_EnumIllegalVal, target));
```

Comment:
`EnumIllegalVal` is `ArgumentException` everywhere. Why change it `ArgumentOutOfRangeException` here?